src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8294,6 +8294,7 @@ private void ParseStatements(ref CSharpSyntaxNode previousNode, SyntaxListBuilde
             {
                 _termState |= TerminatorState.IsSwitchSectionStart;
             }
+            _termState &= ~TerminatorState.IsEndOfDeclarationClause;
```

Comment:
this feles odd/undesriable.  insead of taking us out of a state (i'm really not sure why we'd do that), i'd be curious why we think we're hitting the end of taht state when we encounter one of these tokens.

Reply:
When parsing a local function, this flag has been set in the method responsible for parsing local statements, including variable and function declaration statements. That's the only reason this behavior is specific to local functions.

I find the solution pretty convenient, the place at which the flag is reset seems good to me because there's plenty of method nesting before the real parsing is determined for a local function.

I tried another approach, by evaluating the end result of the termination, and evaluating its terminator state that triggered the termination, and conditionally ignoring it if it was a parenthesis. This approach quickly went down when many tests with valid code suddenly failed. It was too complicated and a very ugly bodge.

Overall I'm pretty confident that this is the approach we want, since I'm removing a flag in a place where the termination state value is being adjusted anyway.

Reply:
@Rekkonnect here's the approach i went with: 

https://github.com/dotnet/roslyn/pull/69482